### PR TITLE
Move to stable Rust 1.65

### PR DESCRIPTION
**Pull request by sunshowers**
It seems like USDT now works on stable Rust, and the one feature used
(`exit_status_error`) has been stabilized as well.

This also addresses most of the clippy issues, but I've broken out the one big nontrivial one into #554.

Depends on #554.

### CODE REVIEW

**Comment by faithanalog** [comment]
yes please

**Review by leftwo — APPROVED** [review]
Yeah, this is all minor changes.

The reason we had been holding off making this change was to keep things compiling
on the mac (as that is my main system).   We had a few times discussed removing that
requirement, but never actually pulled the trigger on it.  It sounds like that time has
finally arrived and it's time for me to let go of local compiling for Crucible.

The mac was never going to be a supported platform, but it was nice while I had it.

File: upstairs/src/lib.rs
Comment (sunshowers) [reply upstairs/src/lib.rs:5530]: Ah no, it was actually clippy --fix that did that, and then I didn't run cargo fmt afterwards (whoops!)

Comment (sunshowers) [reply upstairs/src/lib.rs]: Updated.

**Comment by leftwo** [comment]
Josh is buried in manufacturing stuff, so I would not wait for him to approve, but go ahead and merge this to main.

If you want this in omicron, you will want the git commit after merging to main to use to get the CI produced tar.gz file and SHA hash.   Use those for the package-manifest.toml in Omicron.

**Comment by sunshowers** [comment]
Think I'll skip incorporating this change into omicron for now -- it would require a bit further, and the changes since 861a09c35 aren't relevant to the other repos for now.

**Pull request merged**
